crates/telio-traversal/src/session_keeper.rs

```diff
@@ -62,7 +62,12 @@ pub struct SessionKeeper {
 }
 
 impl SessionKeeper {
-    pub fn start(sock_pool: Arc<SocketPool>) -> Result<Self> {
+    pub fn start(
+        sock_pool: Arc<SocketPool>,
+
+        // This is a way to pass mocked batcher instance when testing
+        #[cfg(test)] batcher: Box<dyn BatcherTrait<PublicKey, State>>,
```

Comment:
Is this a parameter that is only present during testing? And I thought you couldn't "overload" functions in rust 😀  part genius part insanity 🤯

Reply:
You guessed right - `#[cfg(test)]` enables the next expression only when testing `cargo test` :).

The reason I used it boils down to some things:
- I wanted to move batcher to SessionKeeper since it's just a container, same as RepeatedTask(used by SessionKeeper).
- `wg_controller::consolidate()` calls `SessionKeeper` but based on the threshold parameter it might need to call batcher instead of `RepeatedTasks`.
- I want to somehow test that actually batcher was called and the only way to do that is to provide a trait since we use `mockall` that allows mocked traits. This way I could test that the API call went to batcher :)

To be honest there's no harm in leaving out the `cfg(test)` and simply having the parameter always present but I've already seen such pattern several times so I am thinking it might advertise the intent slightly better :)

Reply:
You're totally right to be honest, I was just flexing my freshly gained Rust skills :))